bin/tests/e2e.go

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package tests
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+
+	"github.com/magefile/mage/sh"
+	"github.com/pkg/errors"
+)
+
+var projectName = "node_e2e_test"
+var composeFiles = []string{
+	"bin/localnet/docker-compose.yml",
+	"e2e/docker-compose.yml",
+}
+
+type runner struct {
+	compose         func(args ...string) error
+	etherPassphrase string
+}
+
+// TestE2E runs end-to-end tests
+func TestE2E() error {
+	r := newRunner()
+	defer r.cleanup()
+
+	if err := r.setup(); err != nil {
+		return err
+	}
+
+	return r.test()
+}
+
+func (r *runner) setup() error {
+	fmt.Println("initializing geth node")
+	if err := r.compose("run", "geth", "init", "genesis.json"); err != nil {
+		return errors.Wrap(err, "initializing geth node failed!")
+	}
+	fmt.Println("starting other services")
+	if err := r.compose("up", "-d", "broker", "geth"); err != nil {
+		return errors.Wrap(err, "starting other services failed!")
+	}
+	fmt.Println("starting DB")
+	if err := r.compose("up", "-d", "db"); err != nil {
+		return errors.Wrap(err, "starting DB failed!")
+	}
+
+	for {
+		err := r.compose("exec", "-T", "db", "mysqladmin", "ping", "--protocol=TCP", "--silent")
+		if err != nil {
+			fmt.Print(".")
+		} else {
+			fmt.Println()
+			break
+		}
+	}
+
+	fmt.Println("migrating DB")
```

Comment:
I'd use some logging instead of just prints. At least we will have time marks and levels that will help to read the output.